src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -284,6 +285,22 @@ def _check_config_and_set_final_flags(
                 f" Available names are: {', '.join(self._accelerator_types)}."
             )
 
+        # MPS accelerator is incompatible with DDP family of strategies. It supports single-device operation only.
+        is_ddp_str = isinstance(strategy, str) and "ddp" in strategy
+        is_dp_str = isinstance(strategy, str) and "dp" in strategy
+        is_parallel_strategy = isinstance(strategy, ParallelStrategy) or is_ddp_str or is_dp_str
+        is_mps_accelerator = MPSAccelerator.is_available() and (
+            accelerator in ("mps", "auto", "gpu", None)
+            or isinstance(accelerator, MPSAccelerator)
+            or isinstance(self._strategy_flag, Strategy)
```

Comment:
why this check? wouldn't this be fine if it is a custom single-device strategy or if you really want parallelism and handle that in a custom strategy via CPU?

Reply:
I thought of it as an edge case, but we can also lift this restriction.

Reply:
here or another PR? :)